nat-lab/tests/utils/vm/windows_vm_util.py

```diff
@@ -34,21 +36,22 @@ async def new_connection(ip: str = WINDOWS_1_VM_IP) -> AsyncIterator[Connection]
         compression_algs=None,
     )
 
-    async with asyncssh.connect(
-        ip,
-        username="vagrant",
-        password="vagrant",  # NOTE: this is hardcoded password for transient vm existing only during the tests
-        known_hosts=None,
-        options=ssh_options,
-    ) as ssh_connection:
-        connection = SshConnection(ssh_connection, TargetOS.Windows)
-
-        await _copy_binaries(ssh_connection, connection)
-
-        try:
-            yield connection
-        finally:
-            pass
+    if copy_binaries:
```

Comment:
What happens if `copy_binaries` is False? would this still return a `Connection`?

Reply:
Changed to pytest hook function, `new_connection` always yields a `Connection`